pkg/cmd/ssh/ssh_patch_test.go

```diff
@@ -0,0 +1,386 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ssh_test
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	operationsv1alpha1 "github.com/gardener/gardener/pkg/apis/operations/v1alpha1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	v1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/utils/pointer"
+
+	internalfake "github.com/gardener/gardenctl-v2/internal/fake"
+	gc "github.com/gardener/gardenctl-v2/internal/gardenclient"
+	gcmocks "github.com/gardener/gardenctl-v2/internal/gardenclient/mocks"
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/ssh"
+	"github.com/gardener/gardenctl-v2/pkg/config"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+	targetmocks "github.com/gardener/gardenctl-v2/pkg/target/mocks"
+)
+
+var _ = Describe("SSH Patch Command", func() {
+	const (
+		gardenName           = "mygarden"
+		gardenKubeconfigFile = "/not/a/real/kubeconfig"
+		seedName             = "test-seed"
+		shootName            = "test-shoot"
+	)
+	var (
+		// Subject CN = kubernetes-admin
+		sampleClientCertficate = []byte("-----BEGIN CERTIFICATE-----\nMIIDITCCAgmgAwIBAgIIVteacLIfMJkwDQYJKoZIhvcNAQELBQAwFTETMBEGA1UE\nAxMKa3ViZXJuZXRlczAeFw0yMjExMDQwNzQ1MjVaFw0yMzExMDQwNzQ1MjZaMDQx\nFzAVBgNVBAoTDnN5c3RlbTptYXN0ZXJzMRkwFwYDVQQDExBrdWJlcm5ldGVzLWFk\nbWluMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAwbmPW7NqesUYGtIB\nXOfGrmMeXsWxkM0/sT0LXOZ6c7N/EnRrFAUKMrpMJd0jpePzBN96aLNl08UXpKRj\nHjawqoVEZxcRM6m2sGSeRXCHSz3j1ptkvk2H+fAbJybZ4YvQtT0da1IXxWeBOEvM\nimefH4821KL0MPvOyZLWQaAMZPjvPxqI9nbpOK84KPJ5xGJzOO1lFMC6bNi5QCWG\nypyXWyIjZTsyThu3zNP484NUlE/eixalixGJNKNmMVnzaaytEp/fr0QEzA10HNFV\nemOpsU2r4Yp94GfHpZsmpnUSlUIyWsgFXnJkQNSfHiP4b+5nojwhxTS5KGCiI0cZ\n+ZS82QIDAQABo1YwVDAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUH\nAwIwDAYDVR0TAQH/BAIwADAfBgNVHSMEGDAWgBQd3FiThv+dW40xiwfqbv9bK6Tl\nrzANBgkqhkiG9w0BAQsFAAOCAQEACAVjFMu7htdP9EoHVzZeJIx4SukwDbBeMxIx\nu5hZfQRBDdVTwaias+RiUHeFsyknbcLXvmYRTncr48FJtQozfoYvmFI7AVMBRalO\n7zmhOOsGPKnDDaUWzbo05d4zprMg8x7PWpPJTrnD+ebcNpA+H6K77wg76PaX/vb4\nHUpop7TEo/jsBaaI3AZoJDYXwmij0zjQFUc9E7XZ/p+V345vAIezxolaMHlJZbYv\naolz2dNFv2zTJg1CWkgozWNTY/BaTaJzO4oKVxLRHZWHTrXAU/MRlaRguZQ1tW0f\nS48mSaUhmlEw3l+GE+xjEWe+1pYjZEl2mTNUcOOSMe+hI07E3A==\n-----END CERTIFICATE-----\n")
```

Comment:
I'd rather not hardcode the client certificate, instead I would generate it like so https://github.com/gardener/gardenlogin/blob/c3e7c7cb38fec2854e3d0564d2e14c3ffcde3448/cmd/get_client_certificate_test.go#L480-L493 and serialize it if needed.
This looks like a real client certificate, but probably from a local kind cluster.
If you hardocde one, make sure that the CN etc. clearly indicate that this is a test certificate. Leave other fields empty if not really needed, like the organization, but at least it shouldn't have the special `system:masters` group.